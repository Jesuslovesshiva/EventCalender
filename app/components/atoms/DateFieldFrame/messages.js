import {defineMessages} from 'react-intl'

const scope = 'atoms.date-field-frame'

export default defineMessages({
  toggle: {
    id: `${scope}.toggle`,
    defaultMessage: 'Kalendar ein-/ausschalten',
    description: 'aria label toggle calendar button'
  }
})
