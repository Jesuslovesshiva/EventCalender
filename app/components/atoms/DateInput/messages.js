import {defineMessages} from 'react-intl'

const scope = 'atoms.date-picker'

export default defineMessages({
  ddmmyyyy: {
    id: `${scope}.date`,
    defaultMessage: 'TT.MM.JJJJ',
    description: 'placeholder for format dd.MM.yyyy'
  }
})
