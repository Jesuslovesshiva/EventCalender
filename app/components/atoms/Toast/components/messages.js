import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
