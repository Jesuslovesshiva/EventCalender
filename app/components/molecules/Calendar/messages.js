import {defineMessages} from 'react-intl'

const scope = 'molecules.calendar'

export default defineMessages({
  selectMonth: {
    id: `${scope}.selectMonth`,
    defaultMessage: 'Monat auswählen'
  },
  selectYear: {
    id: `${scope}.selectYear`,
    defaultMessage: 'Jahr auswählen'
  },
  nextMonth: {
    id: `${scope}.nextMonth`,
    defaultMessage: 'nächstes Monat',
    description: 'aria label nextMonth button'
  },
  prevMonth: {
    id: `${scope}.prevMonth`,
    defaultMessage: 'vorheriges Monat',
    description: 'aria label prevMonth button'
  }
})
