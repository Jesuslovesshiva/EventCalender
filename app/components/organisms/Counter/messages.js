import {defineMessages} from 'react-intl'

const scope = 'organisms.counter'

export default defineMessages({
  count: {
    id: `${scope}.count`,
    defaultMessage: 'Zähler: {count}',
    description: 'count output'
  },
  reset: {
    id: `${scope}.reset`,
    defaultMessage: 'Zurücksetzen',
    description: 'reset cta'
  }
})
