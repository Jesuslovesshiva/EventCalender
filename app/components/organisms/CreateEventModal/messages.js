import {defineMessages} from 'react-intl'

const scope = 'organisms.create-event-modal'

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Ereignis hinzuf√ºgen',
    description: 'title of the create-event modal'
  },
  noteLabel: {
    id: `${scope}.noteLabel`,
    defaultMessage: 'Bemerkung',
    description: 'the note input label'
  },
  spaceRequestLabel: {
    id: `${scope}.spaceRequestLabel`,
    defaultMessage: 'Platzwunsch',
    description: 'the space request select label'
  },
  secondSpaceRequestLabel: {
    id: `${scope}.secondSpaceRequestLabel`,
    defaultMessage: 'Platzwunsch 2',
    description: 'the second space request select label'
  },
  spaceLabel: {
    id: `${scope}.spaceLabel`,
    defaultMessage: 'Platz',
    description: 'the assigned space label'
  },
  secondSpaceLabel: {
    id: `${scope}.secondSpaceLabel`,
    defaultMessage: 'Platz 2',
    description: 'the second assigned space label'
  },
  startTimeLabel: {
    id: `${scope}.startTimeLabel`,
    defaultMessage: 'Start',
    description: 'the assigned start time'
  },
  endTimeLabel: {
    id: `${scope}.endTimeLabel`,
    defaultMessage: 'Ende',
    description: 'the assigned end time'
  },
  durationQuickSelect: {
    id: `${scope}.durationQuickSelect`,
    defaultMessage: 'Dauer Schnellauswahl',
    description: 'label for duration quick selection'
  },
  recurringEvent: {
    id: `${scope}.recurringEvent`,
    defaultMessage: 'Serientermin',
    description: 'button text for recurring event'
  },
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Datum',
    description: 'label for date'
  }
})
