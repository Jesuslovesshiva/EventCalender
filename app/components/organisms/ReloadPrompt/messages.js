import {defineMessages} from 'react-intl'

const scope = 'organisms.reload-prompt'

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Neue Inhalte verfügbar',
    description: 'content updates title'
  },
  content: {
    id: `${scope}.content`,
    defaultMessage:
      'Wir haben Neuigkeiten für Sie. Klicken Sie auf Aktualisieren, um die Aktualisierungen zu übernehmen.',
    description: 'tell users to refresh the web-app to apply content updates'
  },
  refresh: {
    id: `${scope}.refresh`,
    defaultMessage: 'Aktualisieren',
    description: 'refresh cta'
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Abbrechen',
    description: 'cancel cta'
  }
})
