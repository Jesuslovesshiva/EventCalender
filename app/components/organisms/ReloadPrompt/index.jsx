import React, {useCallback, useRef} from 'react'
import {FormattedMessage} from 'react-intl'
import {useRegisterSW} from 'virtual:pwa-register/react'

import Button, {BUTTON_TYPES} from '@/components/atoms/Button'
import Prompt from '@/components/molecules/Prompt'

import messages from './messages'

function ReloadPrompt() {
  const {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker
  } = useRegisterSW()

  const confirmButtonRef = useRef()

  const onClose = useCallback(() => {
    setNeedRefresh(false)
  }, [setNeedRefresh])

  const onRefresh = useCallback(() => {
    updateServiceWorker(true)
    onClose()
  }, [updateServiceWorker])

  return (
    <Prompt
      title={<FormattedMessage {...messages.title} />}
      initialFocus={confirmButtonRef}
      isOpen={needRefresh}
      onClose={onClose}
      actions={
        <>
          <Button ref={confirmButtonRef} onClick={onRefresh}>
            <FormattedMessage {...messages.refresh} />
          </Button>
          <Button onClick={onClose} type={BUTTON_TYPES.OUTLINED}>
            <FormattedMessage {...messages.cancel} />
          </Button>
        </>
      }
    >
      <FormattedMessage {...messages.content} />
    </Prompt>
  )
}

export default React.memo(ReloadPrompt)
