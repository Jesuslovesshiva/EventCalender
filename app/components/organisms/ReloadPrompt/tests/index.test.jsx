import {fireEvent} from '@testing-library/react'
import {useRegisterSW} from 'virtual:pwa-register/react'

import render from '@/utils/test-utils/custom-render'

import ReloadPrompt from '../index'
import messages from '../messages'

jest.mock('virtual:pwa-register/react')
jest.mock('@/components/molecules/Prompt')

const renderComponent = () => render(<ReloadPrompt />)

describe('ReloadPrompt', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container).toMatchSnapshot()
  })

  describe('events', () => {
    it('should set needRefresh to false when cancel is clicked', () => {
      const {getByText, queryByText} = renderComponent()
      expect(queryByText(messages.title.defaultMessage)).toBeInTheDocument()
      fireEvent.click(getByText(messages.cancel.defaultMessage))
      expect(queryByText(messages.title.defaultMessage)).not.toBeInTheDocument()
    })

    it('should call the updateServiceWorker function when refresh is clicked', () => {
      const setNeedRefresh = jest.fn()
      const updateServiceWorker = jest.fn()
      useRegisterSW.mockReturnValueOnce({
        needRefresh: [true, setNeedRefresh],
        updateServiceWorker
      })
      const {getByText} = renderComponent()
      fireEvent.click(getByText(messages.refresh.defaultMessage))
      expect(setNeedRefresh).toHaveBeenCalledTimes(1)
      expect(setNeedRefresh).toHaveBeenCalledWith(false)
      expect(updateServiceWorker).toHaveBeenCalledTimes(1)
      expect(updateServiceWorker).toHaveBeenCalledWith(true)
    })
  })
})
