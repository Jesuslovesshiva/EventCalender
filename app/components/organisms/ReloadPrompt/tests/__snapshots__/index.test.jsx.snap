// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadPrompt should match the snapshot 1`] = `
<div>
  <div>
    <div>
      Neue Inhalte verfügbar
    </div>
    <div>
      Wir haben Neuigkeiten für Sie. Klicken Sie auf Aktualisieren, um die Aktualisierungen zu übernehmen.
    </div>
    <div>
      <button
        class="inline-flex justify-center gap-1 font-button-text items-center transition focus-visible:outline-dashed focus-visible:outline-primary focus-visible:outline-offset-1 disabled:pointer-events-none specificity-up:bg-blue-100 min-h-[28px] min-w-[28px] text-white relative overflow-hidden isolate before:bg-gray-dark before:absolute before:inset-0 before:z-[-1] before:opacity-0 before:transition-opacity hover:shadow-md hover:before:opacity-100 disabled:bg-gray-medium disabled:bg-none disabled:text-gray-light disabled:shadow-none shadow bg-gradient-to-l bg-gradient-primary rounded px-3 py-[0.4375rem] "
        type="button"
      >
        Aktualisieren
      </button>
      <button
        class="inline-flex justify-center gap-1 font-button-text items-center transition focus-visible:outline-dashed focus-visible:outline-primary focus-visible:outline-offset-1 disabled:pointer-events-none specificity-up:bg-blue-100 min-h-[28px] min-w-[28px] text-gray-dark border border-gray-medium hover:shadow-md hover:bg-silver hover:border-white disabled:text-gray-medium bg-transparent rounded px-3 py-1.5"
        type="button"
      >
        Abbrechen
      </button>
    </div>
  </div>
</div>
`;
